frontend/javascripts/oxalis/view/right-menu/tree_hierarchy_view_helpers.js

```diff
@@ -86,15 +88,20 @@ export function insertTreesAndTransform(
     const treeNode = makeTreeNodeFromGroup(group, {
       // Ensure that groups are always at the top when sorting by timestamp
       timestamp: 0,
-      expanded: expandedGroupIds[groupId] != null ? expandedGroupIds[groupId] : true,
+      expanded: expandedGroupIds[groupId] != null ? expandedGroupIds[groupId] : false,
       children: insertTreesAndTransform(group.children, groupToTreesMap, expandedGroupIds, sortBy),
     });
     // Groups are always sorted by name and appear before the trees, trees are sorted according to the sortBy prop
     const trees = _.orderBy(groupToTreesMap[groupId] || [], [sortBy], ["asc"]).map(
       makeTreeNodeFromTree,
     );
     treeNode.children = _.orderBy(treeNode.children, ["name"], ["asc"]).concat(trees);
+    treeNode.isIndeterminate = _.some(
+      treeNode.children,
+      groupOrTree => groupOrTree.isChecked || groupOrTree.isIndeterminate,
+    );
     treeNode.isChecked = _.every(treeNode.children, groupOrTree => groupOrTree.isChecked);
+    if (treeNode.isChecked) treeNode.isIndeterminate = false;
```

Comment:
```suggestion
    treeNode.isChecked = _.every(treeNode.children, groupOrTree => groupOrTree.isChecked);
    treeNode.isIndeterminate = treeNode.isChecked ? false : _.some(
      treeNode.children,
      groupOrTree => groupOrTree.isChecked || groupOrTree.isIndeterminate,
    );
```

That way, the `_.some` is only executed when the node is not checked :)

Reply:
Ah, and could that line also be adapted to check for `treeNode.children.length === 0` (not sure if it's an array?) so that empty groups also get the `indeterminate` indicator?

Reply:
> Ah, and could that line also be adapted to check for treeNode.children.length === 0 (not sure if it's an array?) so that empty groups also get the indeterminate indicator?

If I do that, then all groups that contain an empty group (indeterminate) would show up as indeterminate as well. There is a hacky way around that, though, by checking for `groupOrTree.isChecked || (groupOrTree.isIndeterminate && groupOrTree.children.length > 0)`. For a clean solution we would need to save visibility information for groups.

Reply:
There are a couple more edge-cases, like empty groups inside groups with no trees. I'll see whether I can figure out a good solution for this.